### ISSUE TITLE
skaffold is not fully compatible with google artifact registry and container registry URLs

### ISSUE THREAD

**Issue by yuwenma** [GoogleContainerTools/skaffold#4878]
Skaffold uses an external library[ to parse the app image](https://github.com/GoogleContainerTools/skaffold/blob/b4ba29aaba9e443e16b020a2057cfb96e8bc1d57/pkg/skaffold/docker/reference.go#L19). This library could not recognize google artifact registry and container registry images if both tag and digest are given.

### Information

- Skaffold version: v1.14

### Steps to reproduce the behavior
run the following code, the parsing fails with error ”unsupported digest algorithm”. 

```go
package main
 
import (
    "fmt"
	"github.com/docker/distribution/reference"
)
 
func main() {
    r, err := reference.Parse("us-central1-docker.pkg.dev/yuwenma-gke-playground/app/app@sha256:06dd9251e79f06254ec0557945ddb2d7eac5ded1d1c37d4860a24e350a983c52")
	if err != nil {
		fmt.Printf("%v", err)
	}
	if n, ok := r.(reference.Named); ok {
	    fmt.Printf("Image %v\n", n)
	}
	
	if n, ok := r.(reference.Tagged); ok {
	    fmt.Printf("Tag %v\n", n)
	}
 
	if n, ok := r.(reference.Digested); ok {
	    fmt.Printf("Digest %v\n", n)
	}
 
}
```

**Comment by briandealwis** [comment]
I suspect we should use `go-containerregistry`'s `go-containerregistry/pkg/name.ParseReference()`.

**Comment by briandealwis** [comment]
Closing as should be fixed with #5053.  At least your sample code seems to work fine with pkg/skaffold/docker/ParseReference:
```
package main
 
import (
    "fmt"
    "github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
)
 
func main() {
    r, err := docker.ParseReference("us-central1-docker.pkg.dev/yuwenma-gke-playground/app/app@sha256:06dd9251e79f06254ec0557945ddb2d7eac5ded1d1c37d4860a24e350a983c52")
    if err != nil {
	    fmt.Printf("%v", err)
    }
    fmt.Printf("Image %#v\n", r)
}
```
